plugins/org.jboss.tools.openshift.cdk.server/src/org/jboss/tools/openshift/cdk/server/ui/internal/view/CDKActionProvider.java

```diff
@@ -89,8 +84,11 @@ protected Object adaptToViewItem(IServer server) {
 			if( beh != null ) {
 				adb = (ServiceManagerEnvironment)beh.getSharedData(ServiceManagerEnvironment.SHARED_INFO_KEY);
 			}
-			if( adb != null )
-				return new CDKOpenshiftUtility().findExistingOpenshiftConnection(server, adb);
+			
+			if( adb != null ) {
+				OpenshiftUIModel model = new OpenshiftUIModel();
+				return model.getConnectionWrapperForConnection(new CDKOpenshiftUtility().findExistingOpenshiftConnection(server, adb));
```

Comment:
When OpenshiftUIModel instance is created it is added to ConnectionsRegistrySingleton as a listener. When we do not need that instance it should be disposed by model.dispose(). Otherwise it is a memory leak. Moreover in this case it is 'performance leak' as the created connection wrapper will listen to connection changes and build its content. Now, this code may be improved by calling model.dispose(), since it does nothing but removes the listener from the registry. But nobody can predict what may be added to dispose() later.

Guys! I again appeal with creating a singleton instance of OpenshiftUIModel for every client to use. If tests need its own instance let them create it, but lets give the singleton to runtime.